flink-core/src/main/java/org/apache/flink/core/fs/ConsistencyLevel.java

```diff
@@ -0,0 +1,116 @@
+ * call triggering the operation returns, its result is immediately reflected in
+ * in the view presented to any other party calling a file system function.
+ * 
+ * <p>Please note that these levels do not make any statements about the effects or visibility of
+ * file content modification or file appends. In fact, content modification or appending are
+ * not supported in various file systems.
+ * 
+ * <p>Some of these consistency levels indicate that the storage system does not actually
+ * qualify to be called a FileSystem, but rather a blob-/object store.
+ */
+public enum ConsistencyLevel {
+
+	/**
+	 * This consistency level only guarantees that files are visible with a consistent
+	 * view of their contents after their initial creation, once the writing stream has been closed.
+	 * Any modifications, renames, deletes are not guaranteed to be immediately visible in a
+	 * consistent manner.
+	 * 
+	 * <p>To access a file/object consistently, the full path/key must be provided. Enumeration
+	 * of files/objects is not consistent.
+	 * 
+	 * <p>An example of a storage system  with this consistency level is Amazon's S3 object store.
+	 * 
+	 * <p>This is the weakest consistency level with which Flink's checkpointing can work.
+	 * 
+	 * <b>Summary:</b>
+	 * <ul>
+	 *     <li>New file creation: Consistent
+	 *     <li>File deletion: NOT consistent
+	 *     <li>File renaming: NOT consistent
+	 *     <li>Directory listing: NOT consistent
+	 * </ul>
+	 */
+	READ_AFTER_CREATE,
+
+	/**
+	 * This consistency level guarantees that files are visible with a consistent
+	 * view of their contents after their initial creation, and after renaming them.
+	 * The non-existence is consistently visible after delete operations.
+	 *
+	 * <p>To access a file/object consistently, the full path/key must be provided. Enumeration
+	 * of files/objects is not necessarily consistent.
+	 *
+	 * <b>Summary:</b>
+	 * <ul>
+	 *     <li>New file creation: consistent
+	 *     <li>File deletion: consistent
+	 *     <li>File renaming: consistent
+	 *     <li>Directory listing: NOT consistent
+	 * </ul>
+	 */
+	CONSISTENT_RENAME_DELETE,
+	
+	/**
+	 * This consistency level guarantees that files are visible with a consistent
+	 * view of their contents after their initial creation, as well as after renaming operations.
+	 * File deletion is immediately visible to all parties.
+	 * 
+	 * <p>Directory listings are consistent, meaning after file creation/rename/delete, the file
+	 * existence or non-existence is reflected when enumerating the parent directory's contents.
+	 *
+	 * <p>An example of storage systems and file systems falling under this consistency level are
+	 * HDFS, MapR FS, and the Windows file systems.
+	 * 
+	 * <b>Summary:</b>
+	 * <ul>
+	 *     <li>New file creation: consistent
+	 *     <li>File deletion: consistent
+	 *     <li>File renaming: consistent, but not necessarily atomic
```

Comment:
Is atomic renaming guaranteed for `CONSISTENT_RENAME_DELETE`? If not, how important is this atomicity for us? Based on the names one would assume that `LIST_RENAME_DELETE` is all-around "better" than `RENAME_DELETE`, which currently isn't the case and could lead to subtle bugs if this is actually relevant for us.